### PR TITLE
Add a few docblocks to background files

### PR DESCRIPTION
**Pull request by whymarrh**
Refs #3941

This PR adds some of the docblocks for the files listed in #3941:

- [x] `app/scripts/lib/createLoggerMiddleware.js`
- [x] `app/scripts/lib/createOriginMiddleware.js`
- [x] `app/scripts/lib/events-proxy.js`
- [x] `app/scripts/lib/hex-to-bn.js`
- [x] `app/scripts/lib/local-store.js`
- [x] `app/scripts/lib/random-id.js`
- [x] `app/scripts/lib/stream-utils.js`
- [x] `app/scripts/platforms/sw.js`
- [x] `app/scripts/platforms/window.js`

### CODE REVIEW

**Review by danjm — DISMISSED** [review]

**Comment by danjm** [comment]
@whymarrh Will the typescript signatures you've added work with the jsdoc build added in this PR: https://github.com/MetaMask/metamask-extension/pull/3969?

File: app/scripts/lib/stream-utils.js [hunk 1]
```diff
@@ -8,20 +8,34 @@ module.exports = {
...
 function jsonStringifyStream () {
-  return Through.obj(function (obj, encoding, cb) {
+  return Through.obj(function (obj, _, cb) {
     this.push(JSON.stringify(obj))
     cb()
   })
 }
 
+/**
+ * Setups up stream multiplexing for the given stream
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/stream-utils.js]: Thanks

File: app/scripts/lib/createLoggerMiddleware.js [hunk 2]
```diff
@@ -1,14 +1,18 @@
-// log rpc activity
 module.exports = createLoggerMiddleware
 
-function createLoggerMiddleware ({ origin }) {
-  return function loggerMiddleware (req, res, next, end) {
-    next((cb) => {
+/**
+ * Returns a middle that logs RPC activity
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/createLoggerMiddleware.js]: Good catch, thanks

File: app/scripts/lib/createOriginMiddleware.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
-// append dapp origin domain to request
 module.exports = createOriginMiddleware
 
-function createOriginMiddleware ({ origin }) {
-  return function originMiddleware (req, res, next, end) {
-    req.origin = origin
+/**
+ * Returns a middle that appends the DApp origin to request
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/createOriginMiddleware.js]: Good catch, thanks!

File: app/scripts/lib/events-proxy.js [hunk 4]
```diff
@@ -1,26 +1,40 @@
...
+      return (/** @type {any} */ (target))[name]
     },
-    set: (obj, name, value) => {
+    set: (_, name, value) => {
       target[name] = value
       return true
     },
-  })
+  }))
+  /**
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/events-proxy.js]: I think I'll go without one here where this function is closed over, thoughts?

File: app/scripts/lib/hex-to-bn.js [hunk 5]
```diff
@@ -1,6 +1,11 @@
-const ethUtil = require('ethereumjs-util')
+const ethUtil = (/** @type {any} */ (require('ethereumjs-util')))
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/hex-to-bn.js]: It's not correct 😄, good catch! `object` is a stricter value than `any`, as the names suggest, and I've updated this.

[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/hex-to-bn.js]: (Well I guess bother are _correct_ but more strict is better for types.)

File: app/scripts/lib/hex-to-bn.js [hunk 6]
```diff
@@ -1,6 +1,11 @@
...
 const BN = ethUtil.BN
 
+/**
+ * Returns a BinaryNumber representation of the given hex value
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply app/scripts/lib/hex-to-bn.js]: I've linked this to BN

**Comment by whymarrh** [comment]
I've confirmed that these types work with both TypeScript and the JSDoc tooling introduced in #3969

File: app/scripts/lib/local-store.js [hunk 7]
```diff
@@ -1,18 +1,29 @@
...
 const extension = require('extensionizer')
 const log = require('loglevel')
 
+/**
+ * A wrapper around the extension's storage local API
+ *
+ * Note: We should not rely on local storage in an extension!
+ * We should use this instead! [storage.local]{@linkcode https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/local}
+ *
+ * @deprecated
```
Comment (danfinlay) [inline app/scripts/lib/local-store.js]: Actually this file is the `storage.local` wrapper, and is not deprecated.

**Review by danfinlay — APPROVED** [review]
One small thing I would change, otherwise is good.

File: app/scripts/lib/local-store.js (hunk 7 above)
Comment (whymarrh) [reply app/scripts/lib/local-store.js]: Thanks, fixed!

**Pull request merged**